ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionComponent.java

```diff
@@ -235,6 +235,10 @@ public AccessTimeoutDetails getAccessTimeout(Method method) {
     }
 
     public SessionID createSession() {
+        // check to see that StatefulSessionComponent.stop() has not been called concurrent with the request; this can happen on undeploy (see WFLY-5178)
+        if (getCache() == null)
+            throw EjbLogger.ROOT_LOGGER.componentIsShuttingDown();
+
         return this.cache.create().getId();
```

Comment:
Presumably, when there is a concurrent thread invoking StatefulSessionComponent.stop(), which first stops the cache, then sets this property to null.